controllers/dataprotection/restore_controller.go

```diff
@@ -206,16 +206,19 @@ func (r *RestoreReconciler) newAction(reqCtx intctrlutil.RequestCtx, restore *dp
 	repoName, err := CheckBackupRepoForRestore(reqCtx, r.Client, restore)
 	switch {
 	case intctrlutil.IsTargetError(err, intctrlutil.ErrorTypeFatal):
+		dprestore.SetRestoreCheckBackupRepoCondition(restore, dprestore.ReasonCheckBackupRepoFailed, err.Error())
 		restore.Status.Phase = dpv1alpha1.RestorePhaseFailed
 		restore.Status.CompletionTimestamp = &metav1.Time{Time: time.Now()}
 		r.Recorder.Event(restore, corev1.EventTypeWarning, dprestore.ReasonRestoreFailed, err.Error())
 	case intctrlutil.IsTargetError(err, dperrors.ErrorTypeWaitForBackupRepoPreparation):
+		dprestore.SetRestoreCheckBackupRepoCondition(restore, dprestore.ReasonWaitForBackupRepo, err.Error())
 		waitBackupRepo = true
 		restore.Labels[dataProtectionBackupRepoKey] = repoName
 		restore.Labels[dataProtectionWaitRepoPreparationKey] = trueVal
 	case err != nil:
```

Comment:
do not need to set condition in this case?

Reply:
@gnolong May records the error to the condition with `ReasonUnknownError`.